usaspending_api/disaster/v2/views/disaster_base.py

```diff
@@ -26,38 +26,37 @@ def covid_def_code_strings():
 
 def latest_gtas_of_each_year_queryset():
     q = Q()
-    for final_for_fy in finals_for_fy(False):
-        q |= Q(fiscal_year=final_for_fy[0]) & Q(fiscal_period=final_for_fy[1])
-    for final_for_fy in finals_for_fy(True):
-        q |= Q(fiscal_year=final_for_fy[0]) & Q(fiscal_period=final_for_fy[1])
+    for final_for_fy in finals_for_all_fy():
+        q |= Q(fiscal_year=final_for_fy[0]) & Q(fiscal_period=final_for_fy[2])
     if not q:
         return GTASSF133Balances.objects.none()
     return GTASSF133Balances.objects.filter(q)
 
 
 def latest_faba_of_each_year_queryset():
-    q = Q()
-    for final_for_fy in finals_for_fy(False):
-        q |= Q(submission__reporting_fiscal_year=final_for_fy[0]) & Q(
-            submission__reporting_fiscal_period=final_for_fy[1]
-        )
-    for final_for_fy in finals_for_fy(True):
-        q |= Q(submission__reporting_fiscal_year=final_for_fy[0]) & Q(
-            submission__reporting_fiscal_period=final_for_fy[1]
-        )
+    q = filter_by_latest_submissions()
     if not q:
         return FinancialAccountsByAwards.objects.none()
     return FinancialAccountsByAwards.objects.filter(q)
 
 
-def finals_for_fy(is_quarter):
-    return (
-        DABSSubmissionWindowSchedule.objects.filter(
-            submission_reveal_date__lte=datetime.now(timezone.utc), is_quarter=is_quarter
+def filter_by_latest_submissions():
```

Comment:
Would suggest the word "closed" in here: ~`filter_by_latest_closed_submissions`~ or rather `filter_by _latest_closed_periods`, since this is filtering _by_ DABS periods

Reply:
good catch